src/Vo/Database.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+
+namespace App\Vo;
+
+
+use App\Dto\DanceCollection;
+use App\Dto\EntityExtended;
+use App\Entity\Department;
+use App\Entity\Place;
+use App\Entity\Region;
+use App\Entity\Source;
+use App\Entity\Type;
+use App\Entity\Version;
+use Doctrine\ORM\EntityManagerInterface;
+
+class Database
+{
+    /** All versions in database
+     * @var array<int, Version>
+     */
+    private array $versions;
+
+    /** All sources in database
+     * @var array<int, Source>
+     */
+    private array $sources;
+
+    /** All types in database
+     * @var array<int, Type>
+     */
+    private array $types;
+
+    /** All regions in database
+     * @var array<int, Region>
+     */
+    private array $regions;
+
+    /** All departments in database
+     * @var array<int, Department>
+     */
+    private array $departments;
+
+    /** All places in database
+     * @var array<int, Place>
+     */
+    private array $places;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->entityManager = $em;
+        $this->entityManager->getRepository(Version::class)->findAll();
+        $this->sources = $this->entityManager->getRepository(Source::class)->findAll();
+        $this->types = $this->entityManager->getRepository(Type::class)->findAll();
+        $this->regions = $this->entityManager->getRepository(Region::class)->findAll();
+        $this->departments = $this->entityManager->getRepository(Department::class)->findAll();
+        $this->places = $this->entityManager->getRepository(Place::class)->findAll();
```

Comment:
5 all records? For count need use sql: COUNT(*)

Reply:
Then we will need send queries to database for each Type, Source, Region and Department. This not optimise queries to db. Or I don't understand what you mean. I get 5 tables, so I don't need send queries for each Type, Source ...